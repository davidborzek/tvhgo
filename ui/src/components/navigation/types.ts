import { ReactElement } from 'react';

export type INavigationItem = {
  to: string;
  title: string;
  requiredRoles?: string[];
  icon?: ReactElement;
  items?: INavigationItem[];
};
