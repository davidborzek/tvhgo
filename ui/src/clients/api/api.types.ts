export type ListResponse<T> = {
  entries: T[];
  total: number;
  offset: number;
};

export type UserResponse = {
  id: string;
  username: string;
  email: string;
  displayNAme: string;
  createdAt: number;
  updatedAt: number;
};

export type ErrorResponse = {
  message: string;
};

export type EpgEvent = {
  id: number;
  audioDesc: boolean;
  channelId: string;
  channelName: string;
  channelNumber: number;
  piconId: number;
  description: string;
  endsAt: number;
  hd: boolean;
  nextEventId: number;
  startsAt: number;
  subtitle: string;
  subtitled: boolean;
  title: string;
  widescreen: boolean;
  dvrUuid?: string;
  dvrState?: string;
};

export type EpgChannel = {
  channelId: string;
  channelName: string;
  channelNumber: number;
  piconId: number;
  events: EpgEvent[];
};

export type Channel = {
  id: string;
  enabled: boolean;
  name: string;
  number: number;
  piconId: number;
};
