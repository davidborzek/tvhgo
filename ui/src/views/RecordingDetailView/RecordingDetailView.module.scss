.RecordingDetailView {
  padding: 3rem;
  overflow: auto;
}

.cancelButton {
  margin-top: 2rem;
}

.timeForm {
  margin-top: 2rem;
}
