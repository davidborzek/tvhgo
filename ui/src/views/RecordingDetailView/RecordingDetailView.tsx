import { useFormik } from 'formik';
import { useEffect } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import Error from '../../components/Error/Error';
import Input from '../../components/Input/Input';
import {
  useFetchRecording,
  useManageRecordingByEvent,
} from '../../hooks/recording';
import styles from './RecordingDetailView.module.scss';
import * as Yup from 'yup';
import Button from '../../components/Button/Button';
import { useTranslation } from 'react-i18next';
import FormGroup from '../../components/Form/FormGroup/FormGroup';
import EventChannelInfo from '../../components/Event/EventChannelInfo/EventChannelInfo';
import Form from '../../components/Form/Form';
import PairList from '../../components/PairList/PairList';
import Pair from '../../components/PairList/Pair/Pair';
import PairValue from '../../components/PairList/PairValue/PairValue';
import PairKey from '../../components/PairList/PairKey/PairKey';

function RecordingDetailView() {
  const { t } = useTranslation();
  const { id } = useParams();
  const navigate = useNavigate();

  const { recording, error, fetch } = useFetchRecording();
  const {
    cancelRecording,
    stopRecording,
    removeRecording,
    updateRecording,
    pending,
  } = useManageRecordingByEvent();

  const parseStatus = () => {
    switch (recording?.status) {
      case 'scheduled':
        return t('recording_scheduled');
      case 'recording':
        return t('recording_running');
      case 'completed':
        return t('recording_completed');
    }
    return t('recording_unknown');
  };

  const getCancelButtonLabel = () => {
    switch (recording?.status) {
      case 'scheduled':
        return t('cancel_recording');
      case 'recording':
        return t('stop_recording');
      case 'completed':
        return t('delete_recording');
    }
    return '';
  };

  const validationSchema = Yup.object().shape({
    startPadding: Yup.number(),
    endPadding: Yup.number(),
  });

  const handleDeleteOrStop = () => {
    if (recording?.status === 'recording') {
      stopRecording(recording.id, () => {
        fetch(recording.id);
      });
    } else if (recording?.status === 'scheduled') {
      cancelRecording(recording?.id, () => {
        navigate(-1);
      });
    } else if (recording?.status === 'completed') {
      removeRecording(recording.id, () => {
        navigate(-1);
      });
    }
  };

  const formik = useFormik({
    initialValues: {
      startPadding: 0,
      endPadding: 0,
    },
    validationSchema,
    onSubmit: ({ endPadding, startPadding }) => {
      if (!recording) {
        return;
      }

      updateRecording(recording.id, {
        endPadding,
        startPadding,
      });
    },
  });

  useEffect(() => {
    if (id) {
      fetch(id);
    }
  }, [id]);

  useEffect(() => {
    if (recording) {
      formik.setValues({
        startPadding: recording.startPadding,
        endPadding: recording.endPadding,
      });
    }
  }, [recording]);

  const renderCancelButton = () => {
    if (
      recording?.status === 'recording' ||
      recording?.status === 'scheduled' ||
      recording?.status === 'completed'
    ) {
      return (
        <Button
          className={styles.cancelButton}
          type="button"
          label={getCancelButtonLabel()}
          style="red"
          onClick={handleDeleteOrStop}
          disabled={pending}
        />
      );
    }
  };

  const renderTimeForm = () => {
    if (
      recording?.status === 'scheduled' ||
      recording?.status === 'recording'
    ) {
      return (
        <Form
          onSubmit={formik.handleSubmit}
          className={styles.timeForm}
          maxWidth="20rem"
        >
          <FormGroup
            heading={t('recording_time')}
            info={t('recording_duration_padding_info')}
            direction="column"
          >
            <Input
              label={t('recording_minutes_before_start')}
              value={formik.values.startPadding}
              name="startPadding"
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              disabled={recording.status === 'recording' || pending}
              type="number"
            />
            <Input
              label={t('recording_minutes_after_end')}
              value={formik.values.endPadding}
              name="endPadding"
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              type="number"
              disabled={pending}
            />
          </FormGroup>
          <Button type="submit" label={t('save')} disabled={pending} />
        </Form>
      );
    }
  };

  if (error) {
    return <Error message={error} />;
  }

  if (!recording) {
    return <></>;
  }

  return (
    <div className={styles.RecordingDetailView}>
      <EventChannelInfo
        channelName={recording.channelName}
        picon={`/api/picon/${recording.piconId}`}
      />
      <h2>{recording.title}</h2>
      <PairList>
        <Pair>
          <PairKey>{t('subtitle')}</PairKey>
          <PairValue>{recording.subtitle}</PairValue>
        </Pair>
        <Pair>
          <PairKey>{t('airs')}</PairKey>
          <PairValue>{t('event_datetime', { event: recording })}</PairValue>
        </Pair>
        <Pair>
          <PairKey>{t('status')}</PairKey>
          <PairValue>{parseStatus()}</PairValue>
        </Pair>
        <Pair>
          <PairKey>{t('description')}</PairKey>
          <PairValue>{recording.description}</PairValue>
        </Pair>
      </PairList>
      {renderCancelButton()}
      {renderTimeForm()}
    </div>
  );
}

export default RecordingDetailView;
