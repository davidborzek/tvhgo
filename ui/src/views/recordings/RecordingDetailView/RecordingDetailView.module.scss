.RecordingDetailView {
  padding: 3rem;
}

.timeForm {
  margin-top: 2rem;
}

.actions {
  margin-top: 2rem;
  display: flex;
  justify-content: flex-start;
  gap: 1rem;
}
