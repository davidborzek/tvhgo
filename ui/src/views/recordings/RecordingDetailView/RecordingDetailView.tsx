import * as Yup from 'yup';

import {
  LoaderFunctionArgs,
  useLoaderData,
  useNavigate,
  useRevalidator,
} from 'react-router-dom';
import { getRecording, getRecordingUrl } from '@/clients/api/api';

import Button from '@/components/common/button/Button';
import ButtonLink from '@/components/common/button/ButtonLink';
import DeleteConfirmationModal from '@/components/common/deleteConfirmationModal/DeleteConfirmationModal';
import EventChannelInfo from '@/components/epg/event/channelInfo/EventChannelInfo';
import Form from '@/components/common/form/Form';
import FormGroup from '@/components/common/form/FormGroup/FormGroup';
import Input from '@/components/common/input/Input';
import Pair from '@/components/common/pairList/Pair/Pair';
import PairKey from '@/components/common/pairList/PairKey/PairKey';
import PairList from '@/components/common/pairList/PairList';
import PairValue from '@/components/common/pairList/PairValue/PairValue';
import { Recording } from '@/clients/api/api.types';
import styles from './RecordingDetailView.module.scss';
import { useFormik } from 'formik';
import { useManageRecordingByEvent } from '@/hooks/recording';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';

export async function loader({ params }: LoaderFunctionArgs) {
  const { id } = params;
  if (!id) {
    return;
  }

  return getRecording(id);
}

export function Component() {
  const { t } = useTranslation();
  const navigate = useNavigate();
  const { revalidate } = useRevalidator();

  const recording = useLoaderData() as Recording;
  const {
    cancelRecording,
    stopRecording,
    removeRecording,
    updateRecording,
    pending,
  } = useManageRecordingByEvent();

  const [confirmationModalVisible, setConfirmationModalVisible] =
    useState<boolean>(false);

  const parseStatus = () => {
    switch (recording?.status) {
      case 'scheduled':
        return t('recording_scheduled');
      case 'recording':
        return t('recording_running');
      case 'completed':
        return t('recording_completed');
    }
    return t('recording_unknown');
  };

  const getCancelButtonLabel = () => {
    switch (recording?.status) {
      case 'scheduled':
        return t('cancel_recording');
      case 'recording':
        return t('stop_recording');
      case 'completed':
        return t('delete_recording');
    }
    return '';
  };

  const getConfirmationButtonLabel = () => {
    switch (recording?.status) {
      case 'scheduled':
        return t('cancel');
      case 'recording':
        return t('stop');
      case 'completed':
        return t('delete');
    }
    return '';
  };

  const getConfirmationModalTitle = () => {
    switch (recording?.status) {
      case 'scheduled':
        return t('confirm_cancel_recording');
      case 'recording':
        return t('confirm_stop_recording');
      case 'completed':
        return t('confirm_delete_recording');
    }
    return '';
  };

  const validationSchema = Yup.object().shape({
    startPadding: Yup.number(),
    endPadding: Yup.number(),
  });

  const handleDeleteOrStop = () => {
    if (recording?.status === 'recording') {
      stopRecording(recording.id, () => {
        setConfirmationModalVisible(false);
        revalidate();
      });
    } else if (recording?.status === 'scheduled') {
      cancelRecording(recording?.id, () => {
        navigate(-1);
      });
    } else if (recording?.status === 'completed') {
      removeRecording(recording.id, () => {
        navigate(-1);
      });
    }
  };

  const formik = useFormik({
    initialValues: {
      startPadding: recording?.startPadding || 0,
      endPadding: recording?.endPadding || 0,
    },
    validationSchema,
    onSubmit: ({ endPadding, startPadding }) => {
      if (!recording) {
        return;
      }

      updateRecording(recording.id, {
        endPadding,
        startPadding,
      });
    },
    enableReinitialize: true,
  });

  const renderCancelButton = () => {
    if (
      recording?.status === 'recording' ||
      recording?.status === 'scheduled' ||
      recording?.status === 'completed'
    ) {
      return (
        <Button
          type="button"
          label={getCancelButtonLabel()}
          style="red"
          onClick={() => setConfirmationModalVisible(true)}
        />
      );
    }
  };

  const renderDownloadButton = () => {
    if (recording?.status === 'completed') {
      return (
        <ButtonLink
          href={getRecordingUrl(recording.id)}
          download
          label={t('download')}
        />
      );
    }
  };

  const renderTimeForm = () => {
    if (
      recording?.status === 'scheduled' ||
      recording?.status === 'recording'
    ) {
      return (
        <Form
          onSubmit={formik.handleSubmit}
          className={styles.timeForm}
          maxWidth="20rem"
        >
          <FormGroup
            heading={t('recording_time')}
            info={t('recording_duration_padding_info')}
            direction="column"
          >
            <Input
              label={t('recording_minutes_before_start')}
              value={formik.values.startPadding}
              name="startPadding"
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              disabled={recording.status === 'recording' || pending}
              type="number"
              selectTextOnFocus
            />
            <Input
              label={t('recording_minutes_after_end')}
              value={formik.values.endPadding}
              name="endPadding"
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              type="number"
              disabled={pending}
              selectTextOnFocus
            />
          </FormGroup>
          <Button type="submit" label={t('save')} disabled={pending} />
        </Form>
      );
    }
  };

  return (
    <div className={styles.RecordingDetailView}>
      <DeleteConfirmationModal
        visible={confirmationModalVisible}
        onClose={() => setConfirmationModalVisible(false)}
        onConfirm={handleDeleteOrStop}
        title={getConfirmationModalTitle()}
        buttonTitle={getConfirmationButtonLabel()}
        pending={pending}
      />

      <EventChannelInfo
        channelName={recording.channelName}
        picon={`/api/picon/${recording.piconId}`}
      />
      <h2>{recording.title}</h2>
      <PairList>
        <Pair>
          <PairKey>{t('subtitle')}</PairKey>
          <PairValue>{recording.subtitle}</PairValue>
        </Pair>
        <Pair>
          <PairKey>{t('airs')}</PairKey>
          <PairValue>{t('event_datetime', { event: recording })}</PairValue>
        </Pair>
        <Pair>
          <PairKey>{t('status')}</PairKey>
          <PairValue>{parseStatus()}</PairValue>
        </Pair>
        <Pair>
          <PairKey>{t('description')}</PairKey>
          <PairValue>{recording.description}</PairValue>
        </Pair>
      </PairList>

      <div className={styles.actions}>
        {renderCancelButton()}
        {renderDownloadButton()}
      </div>

      {renderTimeForm()}
    </div>
  );
}

Component.displayName = 'RecordingDetailView';
