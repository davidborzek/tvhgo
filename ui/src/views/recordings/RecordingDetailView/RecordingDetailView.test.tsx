import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
import { cleanup, render } from '@testing-library/react';
import { useLoaderData, useNavigate } from 'react-router-dom';

import { Recording } from '@/clients/api/api.types';
import { Component as RecordingDetailView } from './RecordingDetailView';
import { useManageRecordingByEvent } from '@/hooks/recording';
import userEvent from '@testing-library/user-event';

vi.mock('@/hooks/recording');
vi.mock('react-router-dom');

const stopRecordingMock = vi.fn();
const cancelRecordingMock = vi.fn();
const removeRecordingMock = vi.fn();
const updateRecordingMock = vi.fn();

const RECORDING_ID = 'someID';

const navigateMock = vi.fn();

beforeEach(() => {
  vi.mocked(useManageRecordingByEvent).mockReturnValue({
    pending: false,
    cancelRecording: cancelRecordingMock,
    createRecording: async () => {},
    removeRecording: removeRecordingMock,
    stopRecording: stopRecordingMock,
    updateRecording: updateRecordingMock,
  });

  vi.mocked(useNavigate).mockReturnValue(navigateMock);
});

afterEach(() => {
  vi.resetAllMocks();
  cleanup();
});

describe('with status recording', () => {
  test('should render recording', async () => {
    const recording = buildRecording('recording');

    vi.mocked(useLoaderData).mockReturnValue(recording);

    const document = render(<RecordingDetailView />);

    expect(document.asFragment()).toMatchSnapshot();
  });

  test('should stop recording', async () => {
    const recording = buildRecording('recording');

    vi.mocked(useLoaderData).mockReturnValue(recording);

    const document = render(<RecordingDetailView />);

    await userEvent.click(document.getByText('stop_recording'));
    await document.findByText('confirm_stop_recording');

    expect(document.getByText('confirm_stop_recording')).toBeInTheDocument();

    await userEvent.click(document.getByText('stop'));

    expect(stopRecordingMock).toHaveBeenCalledOnce();
  });

  test('should update a recording', async () => {
    const recording = buildRecording('recording');
    vi.mocked(useLoaderData).mockReturnValue(recording);

    const document = render(<RecordingDetailView />);

    const endPaddingInput = document.container.querySelector(
      'input[name="endPadding"]'
    ) as Element;

    const endPadding = 234;

    await userEvent.type(endPaddingInput, `${endPadding}`);

    expect(endPaddingInput).toHaveValue(endPadding);

    await userEvent.click(document.getByText('save'));

    expect(updateRecordingMock).toHaveBeenNthCalledWith(1, recording.id, {
      startPadding: recording.startPadding,
      endPadding,
    });
  });
});

describe('with status completed', () => {
  test('should render recording', async () => {
    const recording = buildRecording('completed');
    vi.mocked(useLoaderData).mockReturnValue(recording);

    const document = render(<RecordingDetailView />);

    expect(document.asFragment()).toMatchSnapshot();
  });

  test('should delete recording', async () => {
    const recording = buildRecording('completed');

    vi.mocked(useLoaderData).mockReturnValue(recording);

    const document = render(<RecordingDetailView />);

    await userEvent.click(document.getByText('delete_recording'));
    await document.findByText('confirm_delete_recording');

    expect(document.getByText('confirm_delete_recording')).toBeInTheDocument();

    await userEvent.click(document.getByText('delete'));

    expect(removeRecordingMock).toHaveBeenCalledOnce();
  });
});

describe('with status scheduled', () => {
  test('should render recording', async () => {
    const recording = buildRecording('scheduled');

    vi.mocked(useLoaderData).mockReturnValue(recording);

    const document = render(<RecordingDetailView />);

    expect(document.asFragment()).toMatchSnapshot();
  });

  test('should cancel recording', async () => {
    const recording = buildRecording('scheduled');

    vi.mocked(useLoaderData).mockReturnValue(recording);

    const document = render(<RecordingDetailView />);

    await userEvent.click(document.getByText('cancel_recording'));
    await document.findByText('confirm_cancel_recording');

    expect(document.getByText('confirm_cancel_recording')).toBeInTheDocument();

    await userEvent.click(document.getByText('cancel'));

    expect(cancelRecordingMock).toHaveBeenCalledOnce();
  });

  test('should update a recording', async () => {
    const recording = buildRecording('scheduled');

    vi.mocked(useLoaderData).mockReturnValue(recording);

    const document = render(<RecordingDetailView />);

    const startPaddingInput = document.container.querySelector(
      'input[name="startPadding"]'
    ) as Element;
    const endPaddingInput = document.container.querySelector(
      'input[name="endPadding"]'
    ) as Element;

    const startPadding = 123;
    const endPadding = 234;

    await userEvent.type(startPaddingInput, `${startPadding}`);
    await userEvent.type(endPaddingInput, `${endPadding}`);

    expect(startPaddingInput).toHaveValue(startPadding);
    expect(endPaddingInput).toHaveValue(endPadding);

    await userEvent.click(document.getByText('save'));

    expect(updateRecordingMock).toHaveBeenNthCalledWith(1, recording.id, {
      startPadding,
      endPadding,
    });
  });
});

const buildRecording = (status: string): Recording => {
  return {
    channelId: 'someChannelId',
    channelName: 'Some Channel',
    createdAt: 0,
    description: 'Some Description',
    duration: 0,
    enabled: true,
    endPadding: 2,
    endsAt: 0,
    eventId: 1,
    extraText: 'Extra Text',
    filename: '',
    id: RECORDING_ID,
    langTitle: {
      ger: 'Some Title',
    },
    originalEndsAt: 0,
    originalStartsAt: 0,
    piconId: 1,
    startPadding: 3,
    startsAt: 0,
    status,
    subtitle: 'Some Subtitle',
    title: 'Some Title',
  };
};
