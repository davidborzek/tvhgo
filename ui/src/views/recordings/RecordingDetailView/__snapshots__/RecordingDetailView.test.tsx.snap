// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`with status completed > should render recording 1`] = `
<DocumentFragment>
  <div
    class="_RecordingDetailView_8b9d52"
  >
    <div
      class="_modalWrapper_12177b "
    >
      <div
        class="_modal_12177b"
        style="max-width: 25rem; width: 100%;"
      >
        <div
          class="_container_12177b"
        >
          <button
            class="_button_2ae682"
          >
            <svg
              fill="none"
              height="800px"
              viewBox="0 0 24 24"
              width="800px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="Menu / Close_LG"
              >
                <path
                  d="M21 21L12 12M12 12L3 3M12 12L21.0001 3M12 12L3 21.0001"
                  id="Vector"
                  stroke="#000000"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </g>
            </svg>
          </button>
          <div
            class="_content_ee1e54"
          >
            <h3
              class="_headline_ee1e54"
            >
              confirm_delete_recording
            </h3>
            <button
              class="_button_a7f9df _medium_a7f9df _red_a7f9df"
              data-testid="confirm-delete-btn"
            >
              delete
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="_EventChannelInfo_e37c34"
    >
      <span
        class="_channelName_e37c34"
      >
        Some Channel
      </span>
      <div
        class="_picon_e37c34"
      >
        <div
          style="width: 100%; height: 100%;"
        />
        <img
          class="_picon_e37c34"
          src="/api/picon/1"
          style="display: none;"
          title="Some Channel"
        />
      </div>
    </div>
    <h2>
      Some Title
    </h2>
    <div
      class="_PairList_768617"
    >
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          subtitle
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          Some Subtitle
        </div>
      </div>
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          airs
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          event_datetime
        </div>
      </div>
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          status
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          recording_completed
        </div>
      </div>
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          description
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          Some Description
        </div>
      </div>
    </div>
    <div
      class="_actions_8b9d52"
    >
      <button
        class="_button_a7f9df _medium_a7f9df _red_a7f9df"
        type="button"
      >
        delete_recording
      </button>
      <a
        class="_button_a7f9df _medium_a7f9df"
        download=""
        href="/api/recordings/someID/stream"
      >
        download
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`with status recording > should render recording 1`] = `
<DocumentFragment>
  <div
    class="_RecordingDetailView_8b9d52"
  >
    <div
      class="_modalWrapper_12177b "
    >
      <div
        class="_modal_12177b"
        style="max-width: 25rem; width: 100%;"
      >
        <div
          class="_container_12177b"
        >
          <button
            class="_button_2ae682"
          >
            <svg
              fill="none"
              height="800px"
              viewBox="0 0 24 24"
              width="800px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="Menu / Close_LG"
              >
                <path
                  d="M21 21L12 12M12 12L3 3M12 12L21.0001 3M12 12L3 21.0001"
                  id="Vector"
                  stroke="#000000"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </g>
            </svg>
          </button>
          <div
            class="_content_ee1e54"
          >
            <h3
              class="_headline_ee1e54"
            >
              confirm_stop_recording
            </h3>
            <button
              class="_button_a7f9df _medium_a7f9df _red_a7f9df"
              data-testid="confirm-delete-btn"
            >
              stop
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="_EventChannelInfo_e37c34"
    >
      <span
        class="_channelName_e37c34"
      >
        Some Channel
      </span>
      <div
        class="_picon_e37c34"
      >
        <div
          style="width: 100%; height: 100%;"
        />
        <img
          class="_picon_e37c34"
          src="/api/picon/1"
          style="display: none;"
          title="Some Channel"
        />
      </div>
    </div>
    <h2>
      Some Title
    </h2>
    <div
      class="_PairList_768617"
    >
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          subtitle
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          Some Subtitle
        </div>
      </div>
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          airs
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          event_datetime
        </div>
      </div>
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          status
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          recording_running
        </div>
      </div>
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          description
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          Some Description
        </div>
      </div>
    </div>
    <div
      class="_actions_8b9d52"
    >
      <button
        class="_button_a7f9df _medium_a7f9df _red_a7f9df"
        type="button"
      >
        stop_recording
      </button>
    </div>
    <form
      class="_Form_27c918 _timeForm_8b9d52"
      style="max-width: 20rem;"
    >
      <div
        class="_FormGroup_d7b0a1"
      >
        <span
          class="_heading_d7b0a1"
        >
          recording_time
        </span>
        <div
          class="_content_d7b0a1 _column_d7b0a1"
        >
          <div
            class="_inputContainer_ae92e9"
          >
            <label
              class="_inputLabel_ae92e9"
              for="startPadding"
            >
              recording_minutes_before_start
            </label>
            <div
              class="_innerContainer_ae92e9"
            >
              <input
                class="_input_ae92e9 _disabled_ae92e9"
                disabled=""
                name="startPadding"
                type="number"
                value="3"
              />
            </div>
          </div>
          <div
            class="_inputContainer_ae92e9"
          >
            <label
              class="_inputLabel_ae92e9"
              for="endPadding"
            >
              recording_minutes_after_end
            </label>
            <div
              class="_innerContainer_ae92e9"
            >
              <input
                class="_input_ae92e9"
                name="endPadding"
                type="number"
                value="2"
              />
            </div>
          </div>
        </div>
        <span
          class="_info_d7b0a1"
        >
          recording_duration_padding_info
        </span>
      </div>
      <button
        class="_button_a7f9df _medium_a7f9df"
        type="submit"
      >
        save
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`with status scheduled > should render recording 1`] = `
<DocumentFragment>
  <div
    class="_RecordingDetailView_8b9d52"
  >
    <div
      class="_modalWrapper_12177b "
    >
      <div
        class="_modal_12177b"
        style="max-width: 25rem; width: 100%;"
      >
        <div
          class="_container_12177b"
        >
          <button
            class="_button_2ae682"
          >
            <svg
              fill="none"
              height="800px"
              viewBox="0 0 24 24"
              width="800px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="Menu / Close_LG"
              >
                <path
                  d="M21 21L12 12M12 12L3 3M12 12L21.0001 3M12 12L3 21.0001"
                  id="Vector"
                  stroke="#000000"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </g>
            </svg>
          </button>
          <div
            class="_content_ee1e54"
          >
            <h3
              class="_headline_ee1e54"
            >
              confirm_cancel_recording
            </h3>
            <button
              class="_button_a7f9df _medium_a7f9df _red_a7f9df"
              data-testid="confirm-delete-btn"
            >
              cancel
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="_EventChannelInfo_e37c34"
    >
      <span
        class="_channelName_e37c34"
      >
        Some Channel
      </span>
      <div
        class="_picon_e37c34"
      >
        <div
          style="width: 100%; height: 100%;"
        />
        <img
          class="_picon_e37c34"
          src="/api/picon/1"
          style="display: none;"
          title="Some Channel"
        />
      </div>
    </div>
    <h2>
      Some Title
    </h2>
    <div
      class="_PairList_768617"
    >
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          subtitle
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          Some Subtitle
        </div>
      </div>
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          airs
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          event_datetime
        </div>
      </div>
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          status
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          recording_scheduled
        </div>
      </div>
      <div
        class="_Pair_2d5b35"
      >
        <div
          class="_PairKey_b66d6a"
        >
          description
        </div>
        <div
          class="_PairValue_05cc8e"
        >
          Some Description
        </div>
      </div>
    </div>
    <div
      class="_actions_8b9d52"
    >
      <button
        class="_button_a7f9df _medium_a7f9df _red_a7f9df"
        type="button"
      >
        cancel_recording
      </button>
    </div>
    <form
      class="_Form_27c918 _timeForm_8b9d52"
      style="max-width: 20rem;"
    >
      <div
        class="_FormGroup_d7b0a1"
      >
        <span
          class="_heading_d7b0a1"
        >
          recording_time
        </span>
        <div
          class="_content_d7b0a1 _column_d7b0a1"
        >
          <div
            class="_inputContainer_ae92e9"
          >
            <label
              class="_inputLabel_ae92e9"
              for="startPadding"
            >
              recording_minutes_before_start
            </label>
            <div
              class="_innerContainer_ae92e9"
            >
              <input
                class="_input_ae92e9"
                name="startPadding"
                type="number"
                value="3"
              />
            </div>
          </div>
          <div
            class="_inputContainer_ae92e9"
          >
            <label
              class="_inputLabel_ae92e9"
              for="endPadding"
            >
              recording_minutes_after_end
            </label>
            <div
              class="_innerContainer_ae92e9"
            >
              <input
                class="_input_ae92e9"
                name="endPadding"
                type="number"
                value="2"
              />
            </div>
          </div>
        </div>
        <span
          class="_info_d7b0a1"
        >
          recording_duration_padding_info
        </span>
      </div>
      <button
        class="_button_a7f9df _medium_a7f9df"
        type="submit"
      >
        save
      </button>
    </form>
  </div>
</DocumentFragment>
`;
